.github/actions/calculate-next-internal-version/action.yml

```diff
@@ -12,15 +12,7 @@ runs:
   using: "composite"
   steps:
     - uses: actions/setup-python@v2
-    - id: get-current-action-branch
-      run: ${{ github.action_path }}/../shared-scripts/get-current-action-branch.sh
-      shell: bash
-    - uses: actions/checkout@v3
-      with:
-        repository: Alfresco/alfresco-build-tools
-        ref: ${{ steps.get-current-action-branch.outputs.action-branch }}
-        path: build-tools
-    - uses: ./build-tools/.github/actions/setup-pysemver
+    - uses: Alfresco/alfresco-build-tools/.github/actions/setup-pysemver@v1.4.0
```

Comment:
Unfortunately, the workaround I used to avoid referencing the version of build-tools inside the project itself is not working in all the scenarios. I.e. it's not working when the cation is invoked by a reusable workflow. The other alternative would systematically use `actions/checkout` to checkout `build-tools` in workflows and then reference the actions using a relative path, but think it will be less intuitive.

Reply:
I still don't have any strong opinion on a solution for this problem that is mainly a lack of flexibility on GHA side.

An alternative to your proposal (checkout + relative paths) that is simplest but not DRY at all is **avoid calling an action inside an action**, that also avoid the problem of actions with a lot of parameters where you would need to support overriding all parameters otherwise caller would be stuck with defaults. That would also enable to use different action versions in the same workflow, don't know if it's a feature or not.

we probably need to experiment a bit and then take a decision on this before we start adding more and more custom actions.

Reply:
It's true that overriding parameters may become cumbersome, but at the same time, I'm not comfortable with repeating the same blocks again and again across several repositories.